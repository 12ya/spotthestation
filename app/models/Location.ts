import { types } from 'mobx-state-tree'
import { Sighting } from './Sightings'

export const Point = types.model('Point', {
  lat: types.optional(types.number, 0),
  lng: types.optional(types.number, 0)
})

export const Location = types.model('Location', {
  title: types.optional(types.string, ''),
  subtitle: types.optional(types.string, ''),
  location: types.optional(Point, ''),
  alert: types.maybeNull(types.boolean),
  sightings: types.optional(types.array(Sighting), []),
})