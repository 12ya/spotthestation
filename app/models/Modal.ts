import { types } from "mobx-state-tree"
export const Modal = types.model("Modal", {
  name: types.optional(types.string, ""),
  state: types.optional(types.string, ""),
})
