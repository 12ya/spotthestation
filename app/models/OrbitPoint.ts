import { types } from "mobx-state-tree"

export const OrbitPoint = types.model("OrbitPoint", {
  date: types.string,
  latitude: types.number,
  longitude: types.number,
  azimuth: types.number,
  elevation: types.number,
})
