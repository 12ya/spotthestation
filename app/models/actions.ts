/* eslint-disable @typescript-eslint/no-unsafe-return */
/* eslint-disable @typescript-eslint/no-floating-promises */
/* eslint-disable @typescript-eslint/no-unsafe-call */
/* eslint-disable @typescript-eslint/no-unsafe-member-access */
/* eslint-disable @typescript-eslint/no-unsafe-argument */
import { flow } from "mobx-state-tree"
import Snackbar from "react-native-snackbar"
import { LocationType } from "../screens/OnboardingScreen/SignupLocation"
import { api } from "../services/api"
import { getCurrentTimeZome } from "../utils/formatDate"
import notifications from "../utils/notifications"
import { Location } from "./Location"
import { Modal } from "./Modal"
import { translate } from "../i18n"

const RootStoreActions = (self) => ({
  getISSSightings: flow(function* getISSSightings(params, isCurrent?: boolean) {
    try {
      const location = isCurrent ? self.currentLocation : self.selectedLocation || self.currentLocation
      const locationCopy = JSON.parse(JSON.stringify(location))
      const { data, ok } = yield api.getISSSightings(params)

      if (ok) {
        const isCurrentLocation = locationCopy.title === self.currentLocation?.title
        const isSelectedLocation = locationCopy.title === self.selectedLocation?.title
        const locationSightings = locationCopy?.sightings ? [...locationCopy?.sightings] : []
        const dataToSave = data.map((item) => {
          const sighting = locationSightings.find(({ date }) => date === item.date)
          return sighting ? { ...item, notify: sighting.notify } : item
        })

        locationCopy.sightings = [...dataToSave]
        if (isSelectedLocation) {
          self.selectedLocation = Location.create(locationCopy)
        }

        let savedLocations = []
        if (!isCurrentLocation) {
          savedLocations = self.savedLocations.filter(({ title }) => title !== locationCopy.title)
          self.savedLocations = [...savedLocations, locationCopy]
        } else {
          self.currentLocation = Location.create(locationCopy)
        }

        const notifyFor = [
          ...(isCurrentLocation ? self.savedLocations : [...savedLocations, locationCopy]),
          isCurrentLocation ? locationCopy : self.currentLocation,
        ]
        if (self.initLoading) self.sightingsLoaded = true
        notifications.setNotifications(notifyFor)
        self.setIsCurrentLocationUpdating(false)
      } else {
        self.trajectoryError = true
        if (self.initLoading) self.sightingsLoaded = true
        self.setIsCurrentLocationUpdating(false)
        Snackbar.show({
          text: data as string,
          duration: Snackbar.LENGTH_LONG,
          action: {
            text: translate('snackBar.dismiss'),
            textColor: "red",
            onPress: () => {
              Snackbar.dismiss()
            },
          },
        })
      }
    } catch (e) {
      self.setIsCurrentLocationUpdating(false)
      console.error(e)
    }
  }),

  setISSSightings: (value: LocationType) => {
    const isCurrentLocation = value.title === self.currentLocation?.title
    const valueCopy = JSON.parse(JSON.stringify(value))
    self.selectedLocation = valueCopy

    let savedLocations = []
    if (!isCurrentLocation) {
      savedLocations = self.savedLocations.filter((location) => location.title !== value.title)
      self.savedLocations = [...savedLocations, valueCopy]
    } else {
      self.currentLocation = Location.create(valueCopy)
    }

    const notifyFor: LocationType[] = [
      ...(isCurrentLocation ? self.savedLocations : [...savedLocations, valueCopy]),
      isCurrentLocation ? valueCopy : self.currentLocation,
    ]

    notifications.setNotifications(notifyFor)
  },

  setCurrentLocation: flow(function* setCurrentLocation(value: LocationType, updateSettingsOnly?: boolean) {
    const valueCopy: LocationType = JSON.parse(JSON.stringify(value))
    if (updateSettingsOnly) {
      self.currentLocation = Location.create(valueCopy)
    } else {
      const { timeZone } = yield getCurrentTimeZome(valueCopy)
      self.currentLocation = Location.create(valueCopy)
      self.getISSSightings({
        zone: timeZone,
        lat: valueCopy.location.lat,
        lon: valueCopy.location.lng,
      }, true)
    }
  }),

  setSelectedLocation: flow(function* setSelectedLocation(value: LocationType) {
    const valueCopy: LocationType = JSON.parse(JSON.stringify(value))
    const { timeZone } = yield getCurrentTimeZome(valueCopy)
    self.selectedLocation = Location.create(valueCopy)
    self.getISSSightings({
      zone: timeZone,
      lat: valueCopy.location.lat,
      lon: valueCopy.location.lng,
    })
  }),

  setSavedLocations: (values: LocationType[]) => {
    self.savedLocations = JSON.parse(JSON.stringify(values))
  },

  setInitLoading: (value: boolean) => {
    self.initLoading = value
  },

  setTrajectoryError: (value: boolean) => {
    self.trajectoryError = value
  },

  setSightingsLoaded: (value: boolean) => {
    self.sightingsLoaded = value
  },

  setIssDataLoaded: (value: boolean) => {
    self.issDataLoaded = value
  },

  setIsCurrentLocationUpdating: (value: boolean) => {
    self.isCurrentLocationUpdating = value
  },

  setNotifications: () => {
    const notifyFor: LocationType[] = [...self.savedLocations, self.currentLocation]

    notifications.setNotifications(notifyFor)
  },

  setNewSavedLocation: flow(function* setNewSavedLocation(value: LocationType) {
    const valueCopy: LocationType = JSON.parse(JSON.stringify(value))
    const { timeZone } = yield getCurrentTimeZome(valueCopy)
    self.savedLocations = [...self.savedLocations, valueCopy]
    const { data, ok } = yield api.getISSSightings({
      zone: timeZone,
      lat: valueCopy.location.lat,
      lon: valueCopy.location.lng,
    })

    if (ok) {
      self.savedLocations = [
        ...self.savedLocations.filter((item) => item.title !== valueCopy.title),
        { ...valueCopy, sightings: [...data] },
      ]

      Snackbar.show({
        text: translate('snackBar.sightingsSaved'),
        duration: Snackbar.LENGTH_LONG,
        action: {
          text: translate('snackBar.dismiss'),
          textColor: "green",
          onPress: () => {
            Snackbar.dismiss()
          },
        },
      })
    } else {
      Snackbar.show({
        text: data as string,
        duration: Snackbar.LENGTH_LONG,
        action: {
          text: translate('snackBar.dismiss'),
          textColor: "red",
          onPress: () => {
            Snackbar.dismiss()
          },
        },
      })
    }
  }),

  getISSData: flow(function* getISSData(params) {
    try {
      const { data, ok } = yield api.getISSData(params)

      if (ok) {
        self.issData = data
        if (self.initLoading) self.issDataLoaded = true
      } else {
        self.trajectoryError = true
        if (self.initLoading) self.issDataLoaded = true
        Snackbar.show({
          text: data as string,
          duration: Snackbar.LENGTH_LONG,
          action: {
            text: translate('snackBar.dismiss'),
            textColor: "red",
            onPress: () => {
              Snackbar.dismiss()
            },
          },
        })
      }
    } catch (e) {
      console.error(e)
    }
  }),

  requestOpenModal: (name: string) => {
    if (self.currentModal && self.currentModal.name === name && self.currentModal.state === "open")
      return

    if (self.currentModal) {
      if (!self.modalsQueue.includes(name)) self.modalsQueue = [...self.modalsQueue, name]
    } else {
      self.currentModal = Modal.create({ name, state: "open" })
    }
  },

  requestCloseModal: (name: string) => {
    if (self.currentModal && self.currentModal.name === name) {
      self.currentModal = Modal.create({ name, state: "closing" })
    } else {
      self.modalsQueue = self.modalsQueue.filter((m) => m !== name)
    }
  },

  closeModal: () => {
    self.currentModal = null
    if (self.modalsQueue.length > 0) {
      self.currentModal = Modal.create({ name: self.modalsQueue[0], state: "open" })
      self.modalsQueue = self.modalsQueue.slice(1)
    }
  },
})

export default RootStoreActions
