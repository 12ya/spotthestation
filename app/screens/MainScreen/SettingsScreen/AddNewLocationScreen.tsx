/* eslint-disable @typescript-eslint/no-unsafe-argument */
/* eslint-disable @typescript-eslint/no-unsafe-member-access */
/* eslint-disable react-native/no-inline-styles */
import { useNavigation, useRoute } from "@react-navigation/native"
import { observer } from "mobx-react-lite"
import React, { useCallback, useRef, useState } from "react"
import { ViewStyle, TextStyle, View } from "react-native"
import { useSafeAreaInsets } from "react-native-safe-area-context"
import Modal from "react-native-modal"
import { Accessory, Button, Icon, Screen, Text, TextField } from "../../../components"
import { colors, spacing, typography } from "../../../theme"
import { LocationType } from "../../OnboardingScreen/SignupLocation"
import { IconLinkButton } from "../../OnboardingScreen/components/IconLinkButton"
import Config from "react-native-config"
import { GooglePlacesAutocomplete, GooglePlacesAutocompleteRef } from "react-native-google-places-autocomplete"
import { translate } from "../../../i18n/translate"
import { RemoveLocationModal } from "./RemoveLocationModal"
import Snackbar from "react-native-snackbar"
import { useStores } from "../../../models"

export interface AddNewLocationScreenParams {
  defaultLocation?: LocationType
}

export const AddNewLocationScreen = observer(function AddNewLocationScreen() {
  const navigation = useNavigation()
  const { savedLocations, setSavedLocations } = useStores()
  const topInset = useSafeAreaInsets().top
  const { params: { defaultLocation } } = useRoute()

  const addressRef = useRef<GooglePlacesAutocompleteRef>()
  const [isFocus, setIsFocus] = useState(false)
  const [isRemove, setIsRemove] = useState(false)
  const [textValue, setTextValue] = useState("")
  const [titleValue, setTitleValue] = useState(defaultLocation?.title || "")
  const [location, setLocation] = useState<LocationType>({...defaultLocation})

  const $headerStyleOverride: TextStyle = {
    top: topInset + 24,
  }

  const handleClear = () => {
    addressRef.current?.clear()
    setTextValue("")
    setLocation(null)
  }

  const handleSave = useCallback(() => {
    location.title = titleValue
    let res = [...savedLocations]
    if (res.find(item => item.title === titleValue)) {
      Snackbar.show({
        text: 'Location with this title already exist!',
        duration: Snackbar.LENGTH_LONG,
        action: {
          text: 'Ok',
          textColor: 'green',
          onPress: () => {
            Snackbar.dismiss()
          },
        },
      })
      return
    }
    
    if (defaultLocation) res = res.filter(item => item.title !== defaultLocation.title)
    res.push(location as any)
    setSavedLocations(res)
    Snackbar.show({
      text: 'Location saved',
      duration: Snackbar.LENGTH_LONG,
      action: {
        text: 'Ok',
        textColor: 'green',
        onPress: () => {
          Snackbar.dismiss()
        },
      },
    })
  }, [titleValue, location, savedLocations])

  const handleRemove = useCallback(() => {
    setSavedLocations(savedLocations.filter(item => item.title !== defaultLocation.title))
    navigation.navigate('LocationSettings' as never, { update: Date.now() } as never)
  }, [defaultLocation, savedLocations])

  return (
    <Screen
      preset="fixed" 
      contentContainerStyle={[$container, $headerStyleOverride]} 
      style={{backgroundColor: colors.palette.neutral900}} 
      statusBarStyle="light"
    >
      <View
        accessible
        accessibilityLabel="Location settings scrollable us area"
        accessibilityHint="Location settings scrollable us area"
        accessibilityRole="scrollbar"
        style={$scrollContainer}
      >
        <View style={$topButtonsContainer}>
          <IconLinkButton icon="x" buttonStyle={$button} iconColor={colors.palette.neutral250} iconSize={20} onPress={() => navigation.navigate('LocationSettings' as never, { update: Date.now() } as never)} />
          {!defaultLocation && <IconLinkButton icon="map" buttonStyle={$button} iconColor={colors.palette.neutral250} iconSize={20} onPress={() => navigation.navigate('AddNewLocationMap' as never)} />}
        </View>
        <Text tx="settings.locationSettingsData.addNewLocation.generalTitle" style={$title} />
        <GooglePlacesAutocomplete
          ref={addressRef}
          placeholder={translate("settings.locationSettingsData.addNewLocation.searchInputPlaceholder")}
          query={{
            key: Config.GOOGLE_API_TOKEN,
            language: 'en',
          }}
          onPress={(data, details = null) => setLocation({ title: details.name, subtitle: details.formatted_address, location: details?.geometry?.location, sightings: [] })}
          onFail={(error) => console.error(error)}
          enablePoweredByContainer={false}
          isRowScrollable={false}
          fetchDetails={true}
          keepResultsAfterBlur={true}
          styles={{
            textInputContainer: isFocus ? [$locations, $active] : $locations,
            textInput: {
              ...$dropdownText,
              ...$dropdownSelected,
            },
            listView: $locationsListContainer,
            row: $locationsRow,
            separator: { height: 0 },
            container: { flex: 0, }
          }}
          textInputProps={{
            placeholderTextColor: colors.palette.neutral450,
            value: location?.subtitle || textValue,
            onFocus: () => setIsFocus(true),
            onBlur: () => setIsFocus(false),
            onChangeText: (text) => setTextValue(text),
            clearButtonMode: "never"
          }}
          renderRow={({ description }) => {
            return <Text text={description} style={$locationsRowText} ellipsizeMode='tail' numberOfLines={1} />
          }}
          renderLeftButton={() => (
            <Icon
              icon="pin"
              size={28}
              color={colors.palette.neutral450}
              containerStyle={$dropdownLeftAccessory}
            />
          )}
          renderRightButton={() => isFocus && textValue && (
            <Icon
              icon="xCircle"
              size={28}
              color={colors.palette.neutral450}
              containerStyle={$dropdownRightAccessory}
              onPress={handleClear}
            />
          )}
        />
        <TextField
          accessible
          accessibilityLabel="location title input"
          accessibilityHint="location title input"
          accessibilityRole="text"
          value={titleValue}
          autoCapitalize="none"
          autoCorrect={false}
          placeholderTx="settings.locationSettingsData.addNewLocation.nameInputPlaceholder"
          onChangeText={setTitleValue}
          inputWrapperStyle={$locations}
          renderLeftAccessory={({ style }) => (
            <Accessory 
              icon="save"
              color={colors.palette.neutral450}
              style={style}
            />
          )}
        />
        <View style={$buttonsContainer}>
          {defaultLocation && <IconLinkButton 
            icon="trash" 
            buttonStyle={$removeButton}
            viewStyle={$removeButton} 
            iconColor={colors.palette.neutral100} 
            iconSize={28} 
            onPress={() => setIsRemove(true)}
          />}
          <Button
            accessible
            accessibilityLabel="save button"
            accessibilityHint="save location"
            tx="settings.locationSettingsData.addNewLocation.saveButton"
            style={[$save, defaultLocation && { width: '80%'}]}
            textStyle={$saveText}
            pressedStyle={$save}
            onPress={handleSave}
          />
        </View>
        <Modal
          isVisible={isRemove}
          onBackdropPress={() => setIsRemove(!isRemove)}
          onSwipeComplete={() => setIsRemove(!isRemove)}
          animationIn="slideInUp"
          animationOut="slideOutDown"
          swipeDirection="down"
          useNativeDriver
          useNativeDriverForBackdrop
          hideModalContentWhileAnimating
          propagateSwipe
          backdropOpacity={0.65}
          style={$modal}
        >
          <RemoveLocationModal onClose={() => setIsRemove(!isRemove)} onRemove={handleRemove} location={defaultLocation} />
        </Modal>
      </View>
    </Screen>
  )
})

const $container: ViewStyle = {
  flex: 1,
  backgroundColor: colors.backgroundDark,
  height: '100%'
}

const $topButtonsContainer: ViewStyle = {
  flexDirection: 'row',
  alignItems: "center",
  justifyContent: "space-between",
  marginBottom: 36
}

const $buttonsContainer: ViewStyle = {
  flexDirection: 'row',
  alignItems: "center",
  justifyContent: "space-between",
}

const $scrollContainer: ViewStyle = { 
  paddingHorizontal: 18,
}

const $button: ViewStyle = {
  backgroundColor: colors.palette.neutral550,
  width: 42,
  height: 42
}

const $text: TextStyle = {
  fontFamily: typography.primary?.normal,
  fontSize: 18,
  lineHeight: 22,
  color: colors.palette.neutral450,
  textAlign: 'left',
  paddingBottom: 24
}

const $title: TextStyle = {
  ...$text,
  fontSize: 36,
  lineHeight: 44,
  color: colors.palette.neutral250,
}

const $locations: ViewStyle = {
  borderWidth: 1.5,
  borderColor: "transparent",
  borderRadius: 28,
  height: 56,
  backgroundColor: colors.palette.neutral350,
  overflow: "hidden",
  marginBottom: 18
}

const $active: ViewStyle = {
  borderWidth: 1.5,
  borderColor: colors.palette.buttonBlue,
  backgroundColor: colors.palette.overlayBlue,
}

const $locationsListContainer: ViewStyle = {
  borderRadius: 12,
  backgroundColor: colors.palette.neutral550,
  overflow: "hidden",
  width: "85%",
  alignSelf: "center",
  marginTop: 3
}

const $dropdownLeftAccessory: ViewStyle = {
  marginStart: spacing.large,
  height: 56,
  justifyContent: "center",
  alignItems: "center",
}

const $dropdownRightAccessory: ViewStyle = {
  marginEnd: spacing.large,
  height: 56,
  justifyContent: "center",
  alignItems: "center",
}

const $dropdownSelected: TextStyle = {
  color: colors.palette.neutral250,
}

const $locationsRow: TextStyle = {
  backgroundColor: 'transparent',
  paddingHorizontal: spacing.large
}

const $locationsRowText: TextStyle = {
  fontSize: 18,
  lineHeight: 22,
  color: colors.palette.neutral250,
  
}

const $dropdownText: TextStyle = {
  flex: 1,
  height: 56,
  alignSelf: "stretch",
  fontFamily: typography.primary.normal,
  fontSize: 18,
  lineHeight: 22,
  paddingHorizontal: 0,
  marginHorizontal: spacing.small,
  textAlignVertical: "center",
  borderRadius: 0,
  backgroundColor: "transparent"
}

const $save: ViewStyle = {
  width: "100%",
  height: 56,
  backgroundColor: colors.palette.buttonBlue,
  borderRadius: 28,
  borderWidth: 0,
  marginVertical: 24
}

const $saveText: TextStyle = {
  color: colors.palette.neutral100,
  fontSize: 18,
  fontFamily: typography.primary.medium,
  lineHeight: 21,
}

const $modal: ViewStyle = {
  flex: 1,
  justifyContent: 'flex-end',
  left: 0,
  margin: 0
}

const $removeButton: ViewStyle = {
  width: 56,
  height: 56,
  backgroundColor: colors.palette.nasaRed,
  alignSelf: 'center'
}
