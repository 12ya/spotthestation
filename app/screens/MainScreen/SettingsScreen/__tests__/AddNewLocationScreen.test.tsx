/* eslint-disable react/display-name */
/* eslint-disable @typescript-eslint/no-unsafe-call */
/* eslint-disable @typescript-eslint/no-unsafe-member-access */
import React from "react"
import renderer from "react-test-renderer"
import { AddNewLocationScreen } from "../AddNewLocationScreen"
import { NavigationContainer } from "@react-navigation/native"
import { fireEvent, render, waitFor } from "@testing-library/react-native"
import { useStores } from "../../../../models"

jest.mock("@react-navigation/native", () => ({
  ...Object.assign({}, jest.requireActual("@react-navigation/native")),
  useRoute: () => ({
    params: {
      id: 1,
      defaultLocation: null,
    },
  }),
}))
it("renders correctly", () => {
  const tree = renderer
    .create(
      <NavigationContainer>
        <AddNewLocationScreen />
      </NavigationContainer>,
    )
    .toJSON()
  expect(tree).toMatchSnapshot()
})

test('clears the text input and sets location value after selecting a place from the list', async () => {
  const { getByPlaceholderText } = render(<NavigationContainer>
    <AddNewLocationScreen />
  </NavigationContainer>)
  const textInput = getByPlaceholderText("settings.locationSettingsData.addNewLocation.searchInputPlaceholder undefined")

  await waitFor(() => {
    expect(textInput.props.value).toBe('')
  })
})

jest.mock('../../../../models', () => ({
  useStores: jest.fn(() => ({
    savedLocations: [],
    setSavedLocations: jest.fn(),
    setNewSavedLocation: jest.fn(),
    handleSave: jest.fn(),
  })),
}))

test('calls handleSave when Save button is clicked', async () => {
  const { getByPlaceholderText } = render(<NavigationContainer>
    <AddNewLocationScreen />
  </NavigationContainer>)
  const { handleSave }: any = useStores()

  const textInput = getByPlaceholderText("settings.locationSettingsData.addNewLocation.searchInputPlaceholder undefined")
  fireEvent.changeText(textInput, 'Some location')

  await waitFor(() => {
    expect(handleSave).toHaveBeenCalledTimes(0)
  })
})
