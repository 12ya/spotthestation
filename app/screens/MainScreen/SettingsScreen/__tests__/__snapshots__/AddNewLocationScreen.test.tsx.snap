// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <View
    accessibilityHint="Location settings scrollable us area"
    accessibilityLabel="Location settings scrollable us area"
    accessibilityRole="scrollbar"
    accessible={true}
    style={
      Object {
        "paddingHorizontal": 26,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "space-between",
          "marginBottom": 52,
        }
      }
    >
      <View
        accessibilityRole="imagebutton"
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            Object {
              "alignItems": "center",
              "backgroundColor": "#1E2022",
              "borderRadius": 143,
              "display": "flex",
              "height": 60,
              "justifyContent": "center",
              "overflow": "hidden",
              "position": "relative",
              "width": 60,
            },
            Object {
              "backgroundColor": "#353739",
              "height": 60,
              "width": 60,
            },
          ]
        }
      >
        <View
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
              },
              Array [
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#1E2022",
                  "borderRadius": 143,
                  "display": "flex",
                  "height": 60,
                  "justifyContent": "center",
                  "overflow": "hidden",
                  "position": "relative",
                  "width": 60,
                },
                undefined,
                undefined,
              ],
            ]
          }
        >
          <ViewManagerAdapter_ExpoBlurView
            proxiedProperties={
              Object {
                "intensity": 0,
                "tint": "default",
              }
            }
            style={
              Object {
                "bottom": 0,
                "left": 0,
                "position": "absolute",
                "right": 0,
                "top": 0,
              }
            }
          />
          <View>
            <RCTImageView
              internal_analyticTag={null}
              resizeMode="contain"
              source={
                Array [
                  Object {
                    "height": undefined,
                    "uri": undefined,
                    "width": undefined,
                  },
                ]
              }
              style={
                Object {
                  "height": 29,
                  "overflow": "hidden",
                  "resizeMode": "contain",
                  "tintColor": "#DEDDDE",
                  "width": 29,
                }
              }
              tintColor="#DEDDDE"
            />
          </View>
        </View>
      </View>
      <View
        accessibilityRole="imagebutton"
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            Object {
              "alignItems": "center",
              "backgroundColor": "#1E2022",
              "borderRadius": 143,
              "display": "flex",
              "height": 60,
              "justifyContent": "center",
              "overflow": "hidden",
              "position": "relative",
              "width": 60,
            },
            Object {
              "backgroundColor": "#353739",
              "height": 60,
              "width": 60,
            },
          ]
        }
      >
        <View
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
              },
              Array [
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#1E2022",
                  "borderRadius": 143,
                  "display": "flex",
                  "height": 60,
                  "justifyContent": "center",
                  "overflow": "hidden",
                  "position": "relative",
                  "width": 60,
                },
                undefined,
                undefined,
              ],
            ]
          }
        >
          <ViewManagerAdapter_ExpoBlurView
            proxiedProperties={
              Object {
                "intensity": 0,
                "tint": "default",
              }
            }
            style={
              Object {
                "bottom": 0,
                "left": 0,
                "position": "absolute",
                "right": 0,
                "top": 0,
              }
            }
          />
          <View>
            <RCTImageView
              internal_analyticTag={null}
              resizeMode="contain"
              source={
                Array [
                  Object {
                    "height": undefined,
                    "uri": undefined,
                    "width": undefined,
                  },
                ]
              }
              style={
                Object {
                  "height": 29,
                  "overflow": "hidden",
                  "resizeMode": "contain",
                  "tintColor": "#DEDDDE",
                  "width": 29,
                }
              }
              tintColor="#DEDDDE"
            />
          </View>
        </View>
      </View>
    </View>
    <Text
      allowFontScaling={false}
      style={
        Array [
          Object {},
          Array [
            Object {
              "fontSize": 23,
              "lineHeight": 35,
            },
            Object {
              "fontFamily": "Helvetica Neue",
            },
            Object {
              "color": "#191015",
            },
          ],
          undefined,
          undefined,
          Object {
            "color": "#DEDDDE",
            "fontFamily": "Helvetica Neue",
            "fontSize": 52,
            "lineHeight": 63,
            "paddingBottom": 35,
            "textAlign": "left",
          },
        ]
      }
    >
      settings.locationSettingsData.addNewLocation.generalTitleAdd undefined
    </Text>
    <View
      pointerEvents="box-none"
      style={
        Array [
          Object {
            "flex": 1,
          },
          Object {
            "flex": 0,
          },
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "flexDirection": "row",
            },
            Object {
              "backgroundColor": "#1E2022",
              "borderColor": "transparent",
              "borderRadius": 40,
              "borderWidth": 2,
              "height": 80,
              "marginBottom": 26,
              "overflow": "hidden",
            },
          ]
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "height": 80,
              "justifyContent": "center",
              "marginStart": 35,
            }
          }
        >
          <RCTImageView
            internal_analyticTag={null}
            resizeMode="contain"
            source={
              Array [
                Object {
                  "height": undefined,
                  "uri": undefined,
                  "width": undefined,
                },
              ]
            }
            style={
              Object {
                "height": 40,
                "overflow": "hidden",
                "resizeMode": "contain",
                "tintColor": "#ADADAE",
                "width": 40,
              }
            }
            tintColor="#ADADAE"
          />
        </View>
        <TextInput
          allowFontScaling={false}
          clearButtonMode="never"
          onBlur={[Function]}
          onChangeText={[Function]}
          onFocus={[Function]}
          placeholder="settings.locationSettingsData.addNewLocation.searchInputPlaceholder undefined"
          placeholderTextColor="#ADADAE"
          style={
            Array [
              Object {
                "backgroundColor": "#FFFFFF",
                "borderRadius": 5,
                "flex": 1,
                "fontSize": 15,
                "height": 44,
                "marginBottom": 5,
                "paddingHorizontal": 10,
                "paddingVertical": 5,
              },
              Object {
                "alignSelf": "stretch",
                "backgroundColor": "transparent",
                "borderRadius": 0,
                "color": "#DEDDDE",
                "flex": 1,
                "fontFamily": "Helvetica Neue",
                "fontSize": 26,
                "height": 80,
                "lineHeight": 32,
                "marginHorizontal": 17,
                "paddingHorizontal": 0,
                "textAlignVertical": "center",
              },
            ]
          }
          value=""
        />
      </View>
    </View>
    <View
      accessibilityState={
        Object {
          "disabled": false,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "flex-start",
              "backgroundColor": "#1E2022",
              "borderRadius": 40,
              "flexDirection": "row",
              "height": 80,
              "overflow": "hidden",
            },
            false,
            undefined,
            Object {
              "paddingStart": 0,
            },
            undefined,
            Object {
              "backgroundColor": "#1E2022",
              "borderColor": "transparent",
              "borderRadius": 40,
              "borderWidth": 2,
              "height": 80,
              "marginBottom": 26,
              "overflow": "hidden",
            },
          ]
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "height": 80,
              "justifyContent": "center",
              "marginStart": 35,
            }
          }
        >
          <RCTImageView
            internal_analyticTag={null}
            resizeMode="contain"
            source={
              Array [
                Object {
                  "height": undefined,
                  "uri": undefined,
                  "width": undefined,
                },
              ]
            }
            style={
              Object {
                "height": 40,
                "overflow": "hidden",
                "resizeMode": "contain",
                "tintColor": "#ADADAE",
                "width": 40,
              }
            }
            tintColor="#ADADAE"
          />
        </View>
        <TextInput
          accessibilityHint="location title input"
          accessibilityLabel="location title input"
          accessibilityRole="text"
          accessible={true}
          allowFontScaling={false}
          autoCapitalize="none"
          autoCorrect={false}
          editable={true}
          onChangeText={[Function]}
          placeholder="settings.locationSettingsData.addNewLocation.nameInputPlaceholder undefined"
          placeholderTextColor="#ADADAE"
          style={
            Array [
              Object {
                "alignSelf": "stretch",
                "color": "#DEDDDE",
                "flex": 1,
                "fontFamily": "Helvetica Neue",
                "fontSize": 26,
                "height": 80,
                "marginHorizontal": 17,
                "paddingHorizontal": 0,
                "paddingVertical": 0,
                "textAlignVertical": "center",
              },
              false,
              undefined,
              undefined,
              undefined,
            ]
          }
          textAlignVertical="top"
          underlineColorAndroid="rgba(0, 0, 0, 0)"
          value=""
        />
      </View>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "space-between",
        }
      }
    >
      <View
        accessibilityHint="save location"
        accessibilityLabel="save button"
        accessibilityRole="button"
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            Array [
              Object {
                "alignItems": "center",
                "borderRadius": 6,
                "flexDirection": "row",
                "justifyContent": "center",
                "minHeight": 80,
                "overflow": "hidden",
                "paddingHorizontal": 17,
                "paddingVertical": 17,
              },
              Object {
                "backgroundColor": "#FFFFFF",
                "borderColor": "#B6ACA6",
                "borderWidth": 1,
              },
            ],
            Object {
              "backgroundColor": "#0B3D91",
              "borderRadius": 40,
              "borderWidth": 0,
              "height": 80,
              "marginVertical": 35,
              "width": "100%",
            },
            false,
          ]
        }
        testID="saveButton"
      >
        <Text
          allowFontScaling={false}
          style={
            Array [
              Object {},
              Array [
                Object {
                  "fontSize": 23,
                  "lineHeight": 35,
                },
                Object {
                  "fontFamily": "Helvetica Neue",
                },
                Object {
                  "color": "#191015",
                },
              ],
              undefined,
              undefined,
              Array [
                Object {
                  "flexGrow": 0,
                  "flexShrink": 1,
                  "fontFamily": "HelveticaNeue-Medium",
                  "fontSize": 23,
                  "lineHeight": 29,
                  "textAlign": "center",
                  "zIndex": 2,
                },
                Object {
                  "color": "#FFFFFF",
                  "fontFamily": "HelveticaNeue-Medium",
                  "fontSize": 26,
                  "lineHeight": 30,
                },
                false,
              ],
            ]
          }
        >
          settings.locationSettingsData.addNewLocation.saveButton undefined
        </Text>
      </View>
    </View>
  </View>
</div>
`;
